service/src/main/java/uk/gov/hmcts/reform/fpl/service/removeorder/GeneratedOrderRemovalAction.java

```diff
@@ -0,0 +1,82 @@
+package uk.gov.hmcts.reform.fpl.service.removeorder;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.interfaces.RemovableOrder;
+import uk.gov.hmcts.reform.fpl.model.order.generated.GeneratedOrder;
+import uk.gov.hmcts.reform.fpl.utils.CaseDetailsMap;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+
+@Component
+public class GeneratedOrderRemovalAction implements OrderRemovalAction {
+
+    @Override
+    public boolean isAccepted(RemovableOrder removableOrder) {
+        return removableOrder instanceof GeneratedOrder;
+    }
+
+    @Override
+    public void remove(CaseData caseData,
+                       CaseDetailsMap data, UUID removedOrderId,
+                       RemovableOrder removableOrder) {
+
+        GeneratedOrder generatedRemovableOrder = (GeneratedOrder) removableOrder;
+
+        List<Element<GeneratedOrder>> generatedOrders = caseData.getOrderCollection();
+        boolean removed = generatedOrders.remove(element(removedOrderId, generatedRemovableOrder));
+
+        if (!removed) {
+            throw new IllegalArgumentException(format("Failed to find order matching id %s", removedOrderId));
```

Comment:
Do you think it would be nicer to make an order or removable order not found exception